repos/spack_repo/builtin/packages/octave/package.py

```diff
@@ -86,6 +87,7 @@ class Octave(AutotoolsPackage, GNUMirrorPackage):
     # Required dependencies
     depends_on("c", type="build")
     depends_on("cxx", type="build")
+    depends_on("cxx@17:", when="@10:")
```

Comment:
Let's not add this, since this is bound to fail when other packages will require e.g. `cxx@14` and are mixed in the same DAG. We discussed sometimes ago that the best way to deal with this is to extend virtual packages with variants, but this is not implemented yet.

Reply:
OK, removed.

Reply:
Thanks!